### PR TITLE
David ai implementation

### PR DESCRIPTION
**Pull request by david-the-droid**
Creating pull request to pull changes across to the base repo for AI player implementation for oxo game. @grahame-student 

Changes made: 

* Added a minimax function implementation for AI player
* Added a check win function for the grid the AI player uses to perform analysis
* Added detailed comments for current implementation and operation in code
* Added some functionality as a work around should the AI player be called first or second in a a match (tweaks minimax verify win logic)

### CODE REVIEW

File: example.py.bak [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        result_line = f"{player1.name:12}"
+        for player2 in player_list:
+            values = matrix[player1.id][player2.id]
+            scores = ",".join([str(i) for i in values])
+            result_line += f"{scores:12} "
+        print(result_line)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (grahame-student) [inline example.py.bak:64]: No need for a backup file, we have version control

File: example.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 from oxo_tourney.player import RandomPlayer
 from oxo_tourney.tourney import Tourney
 
+#
```
Comment (grahame-student) [inline example.py:5]: Was this the only change? Is it required?

**Review by grahame-student — CHANGES_REQUESTED** [review]

File: example.py (hunk 2 above)
Comment (david-the-droid) [reply example.py:5]: Hi Grahame, this was me testing github out with a quick change which needs to be removed

File: example.py.bak (hunk 1 above)
Comment (david-the-droid) [reply example.py.bak]: Ah didn't release a back up of the file had been made with the commit. removed the file from pull request and my branch

File: example.py (hunk 2 above)
Comment (david-the-droid) [reply example.py]: Deleted "#" and change should commit to the david-ai-implementation and come through to the pull request

**Review by grahame-student — APPROVED** [review]

**Comment by grahame-student** [comment]
@david-the-droid You've not modified the __init__.py file in the player namespace.

**Pull request merged**